src/parser.ts

```diff
@@ -414,30 +425,74 @@ function parseAnySwitchFromToolArguments(args: string, excludeArgs: string[]): s
         }
 
         if (!exclude) {
-            // The other parser helpers differ from this one by the fact that they know
-            // what switch they are looking for. This helper first identifies anything
-            // that looks like a switch and then calls parseMultipleSwitchFromToolArguments
-            // which knows how to parse complex scenarios of spaces, quotes and other characters.
-            // Don't allow parseMultipleSwitchFromToolArguments to remove surrounding quotes for switch values.
-            let swiValues: string[] = parseMultipleSwitchFromToolArguments(partialArgs, swi, false);
-
-            // If no values are found, it means the switch has simple syntax.
-            // Add this to the array.
-            if (swiValues.length === 0) {
-                swiValues.push(swi);
-            }
+            compilerArgRegions += partialArgs;
+        }
+
+        match1 = match2;
+    }
+
+    let parseCompilerArgsScriptFile: string = util.parseCompilerArgsScriptFile();
+    let parseCompilerArgsScriptContent: string;
+    if (process.platform === "win32") {
+        // There is a potential problem with the windows version of the script:
+        // A fragment like "-sw1,-sw2,-sw3" gets split by comma and a fragment like
+        // "-SwDef=Val" is split by equal. Opened GitHub issue
+        // https://github.com/microsoft/vscode-makefile-tools/issues/149.
+        // For now, these scenarios don't happen on windows (the comma syntax is linux only
+        // and the equal syntax is encountered for defines and c/c++ standard which are parsed
+        // separately, not via this script invocation).
+        parseCompilerArgsScriptContent = `@echo off\r\n`;
+        parseCompilerArgsScriptContent += `for %%i in (%*) do echo %%i \r\n`;
+    } else {
+        parseCompilerArgsScriptContent = `#!/bin/bash \n`;
+        parseCompilerArgsScriptContent += `while (( "$#" )); do \n`;
+        parseCompilerArgsScriptContent += `  echo $1 \n`;
+        parseCompilerArgsScriptContent += `  shift \n`;
+        parseCompilerArgsScriptContent += `done \n`;
+    }
+
+    // Create the script only when not found. The extension makes sure to delete it regularly
+    // (at project load time).
+    if (!util.checkFileExistsSync(parseCompilerArgsScriptFile)) {
+        util.writeFile(parseCompilerArgsScriptFile, parseCompilerArgsScriptContent);
+    }
+
+    let scriptArgs: string[] = [];
+    let runCommand: string;
+    if (process.platform === 'win32') {
+        runCommand = "cmd";
+        scriptArgs.push("/c");
+        scriptArgs.push(parseCompilerArgsScriptFile);
```

Comment:
Should we surround the file name with double quotes or do you guarantee no spaces in the file name?

Reply:
similar question for the *nix invocation - might need single quotes? (I forget the rules there)

Reply:
ok, so the way this is called, you might not need it on windows.  But I still have a concern about linux.

Reply:
Yes a file with spaces was not able to work. I fixed on linux with single quotes around file name but on windows it's broken too, but adding double quotes around file name is not helping. Investigating how to fix on windows too.
Luckily, the extension is not generating name with space but if TEMP happens to be defined to a path with space then we would have this problem. Thanks for thinking of this scenario.

Reply:
I fixed windows as well (I was sending parameters of script as parameters of cmd.exe and confused it).
I tested these diffs again on both.